polar-core/src/walker.rs

```diff
@@ -0,0 +1,290 @@
+use super::partial::Constraints;
+use super::rules::*;
+use super::terms::*;
+
+pub trait Visitor<'term>: Sized {
```

Comment:
I'm not sure these `'term` lifetimes are currently necessary I was able to get rid of them without any problems. They would be necessary if you want to store references to terms on the visitor (I think), but I don't see anywhere we do that.

Reply:
Removed in 47a84bf